### DISCUSSION TITLE
React Form Hook with UI5 Web Components React

### DISCUSSION THREAD

**Discussion by apoorvsemwal** [react-hook-form/react-hook-form#6068] [Q&A] [answered]
Hello,
I am just starting to use RFH with [ui5-webcomponents-react](https://sap.github.io/ui5-webcomponents-react/?path=/story/getting-started--page) and was hoping if somebody could help me understand why my console log is not working for a dummy component defined here in this codesnadbox:
[CodeSandBox](https://codesandbox.io/s/react-hook-form-with-ui5-web-components-react-filter-bar-1enqx)
This has 2 components:
**BasicForm** - Works as expected and I can see the form data in console log
**UI5FilterBar** - handleSubmit crashes with e.persist is not a function - I understand somehow wrapping my FilterBar inside a form might not be the right way to make it work. It would be a great help if somebody could help me understand the right way to make it work with [UI5 Web component React FilterBar](https://sap.github.io/ui5-webcomponents-react/?path=/docs/layouts-floorplans-filterbar--default-story)  -  [Github for FilterBar](https://github.com/SAP/ui5-webcomponents-react/tree/main/packages/main/src/components/FilterBar)

Thanks

**Comment by MarcusNotheis** [comment, marked as answer]
I guess this should have been fixed with versions >=7.2.0 (https://github.com/react-hook-form/react-hook-form/pull/4825). I tried to go into a bit more detail at https://github.com/SAP/ui5-webcomponents-react/discussions/1865.

**Reply by apoorvsemwal** [reply]
Hello @MarcusNotheis
I tried upgrading the RHF version based on your explanation and it did work out well for me.
Thanks.

**Reply by bluebill1049** [reply]
thank you @MarcusNotheis
